src/Storages/ExternalStream/Kafka/KafkaSource.h

```diff
@@ -62,9 +62,12 @@ class KafkaSource final : public ISource
     Poco::Logger * log;
 
     Block header;
+    const Block non_virtual_header;
```

Comment:
This is only used for getting the input format, no need to make it a member variable?

Reply:
I didn't notice it's also used in `parseFormat`.